lib/compat/wordpress-6.5/blocks.php

```diff
@@ -99,16 +95,16 @@ function gutenberg_process_block_bindings( $block_content, $block, $block_instan
 				continue;
 			}
 			// If no source is provided, or that source is not registered, process next attribute.
-			if ( ! isset( $binding_source['source'] ) || ! isset( $binding_source['source']['name'] ) || ! isset( $block_bindings_sources[ $binding_source['source']['name'] ] ) ) {
+			if ( ! isset( $binding_source['source'] ) || ! is_string( $binding_source['source'] ) || ! isset( $block_bindings_sources[ $binding_source['source'] ] ) ) {
 				continue;
 			}
 
-			$source_callback = $block_bindings_sources[ $binding_source['source']['name'] ]['apply'];
+			$source_callback = $block_bindings_sources[ $binding_source['source'] ]['apply'];
```

Comment:
I think that we might need to introduce `WP_Block_Binding_Source` class so we don't have to do so many checks in the code. Here, we should check whether `apply` is defined with `asset` since it might not be as we don't don any validation at all.

Reply:
This could make sense 🤔 I've added that as a follow-up task in [the tracking issue](https://github.com/WordPress/gutenberg/issues/53300).